### PR TITLE
colour: add a fallback for greyscale profile

### PR DESCRIPTION
**Pull request by kleisauke**
Uses the CC0-licensed 'sGrey-v2-magic.icc' profile from https://github.com/saucecontrol/Compact-ICC-Profiles#greyscale.

Ensures that greyscale images with an incompatible ICC profile has an appropriate fallback profile.

---
Test case:
```console
$ vips black invalid-icc.png 1 1
$ exiftool "-icc_profile<=libvips/colour/profiles/sRGB.icm" invalid-icc.png
    1 image files updated
```
    
Before:
```console
$ vipsthumbnail invalid-icc.png -e srgb

(vipsthumbnail:11620): VIPS-WARNING **: 12:57:17.408: profile incompatible with image

(vipsthumbnail:11620): VIPS-WARNING **: 12:57:17.408: profile incompatible with image
vipsthumbnail: unable to thumbnail invalid-icc.png
icc_transform: unable to load or find any compatible input profile
```

After:
```console
$ vipsthumbnail invalid-icc.png -e srgb

(vipsthumbnail:11693): VIPS-WARNING **: 12:57:47.791: profile incompatible with image
```

> **Note**: this PR targets the [`8.14`](https://github.com/libvips/libvips/tree/8.14) branch.

### CODE REVIEW

File: libvips/colour/icc_transform.c [hunk 1]
```diff
@@ -720,8 +720,26 @@ vips_icc_set_import( VipsIcc *icc,
...
+		case VIPS_INTERPRETATION_GREY16:
+			name = "sgrey";
+			break;
+
+		case VIPS_INTERPRETATION_CMYK:
+			name = "cmyk";	
+			break;
+
+		case VIPS_INTERPRETATION_RGB16:
+			name = "p3";
```
Comment (jcupitt) [inline libvips/colour/icc_transform.c:736]: Is this a good idea? It would meaning converting an RGB image to RGB16 could change the colour drastically.

Comment (kleisauke) [reply libvips/colour/icc_transform.c]: Ah, I forgot to omit this change. This should be part of another PR.

IIUC, this path is only taken for RGB16 -> RGB and not vice versa, but perhaps I misunderstood.

**Review by jcupitt — APPROVED** [review]
LGTM!

**Comment by kleisauke** [comment]
Commit aecc8793800e832a627ac9b1559957f55ee4e9c1 uses the ICC v2 variant instead, since the v4 profile isn't spec-compliant, see: https://github.com/libvips/libvips/pull/3400#issuecomment-1479518511.

**Pull request merged**